specification/dnc/resource-manager/Microsoft.DelegatedNetwork/preview/2020-08-08-preview/DelegatedNetwork.json

```diff
@@ -0,0 +1,820 @@
+          }
+        },
+        "tags": [
+          "Controller"
+        ],
+        "operationId": "Controller_Create",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/controllerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "controllerParameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ControllerTypeParameters"
+            },
+            "description": "controller type parameters"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/DelegatedController"
+            }
+          },
+          "201": {
+            "description": "InProgress. The operation is still in progress.",
+            "schema": {
+              "$ref": "#/definitions/DelegatedController"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "delete": {
+        "description": "Deletes the DNC controller",
+        "x-ms-examples": {
+          "Delete controller": {
+            "$ref": "./examples/deleteController.json"
+          }
+        },
+        "tags": [
+          "Controller"
+        ],
+        "operationId": "Controller_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/controllerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK."
+          },
+          "202": {
+            "description": "Accepted."
+          },
+          "204": {
+            "description": "No Content."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{vnetName}/subnets/{subnetName}/providers/Microsoft.DelegatedNetwork/delegatedSubnet": {
```

Comment:
this path wont work with ARM. You need a {name} segment to indicate the name of the delegatedSubnet

Reply:
and then add a GET api to list all delegatedSubnets under a subnet.

Reply:
pluralize the resource type name so delegatedSubnets

Reply:
> this path wont work with ARM. You need a {name} segment to indicate the name of the delegatedSubnet

can we give same subnet name?

Reply:
> and then add a GET api to list all delegatedSubnets under a subnet.

subnet is same here and the only thing is its delegated. so there won't be any list of delegated subnets under subnet